exp/api/v1alpha1/azureasomanagedmachinepool_types.go

```diff
@@ -20,8 +20,13 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// AzureASOManagedMachinePoolKind is the kind for AzureASOManagedMachinePool.
-const AzureASOManagedMachinePoolKind = "AzureASOManagedMachinePool"
+const (
+	// AzureASOManagedMachinePoolKind is the kind for AzureASOManagedMachinePool.
+	AzureASOManagedMachinePoolKind = "AzureASOManagedMachinePool"
+
+	// AzureASOManagedMachinePoolFinalizer is the finalizer added to AzureASOManagedMachinePools.
+	AzureASOManagedMachinePoolFinalizer = "azureASOManagedMachinePool.infrastructure.cluster.x-k8s.io"
```

Comment:
It looks like other finalizers in CAPZ follow a convention of all lower-case characters. Not sure if this is required but just noting it for consistency: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/azurecluster_types.go#L27

Reply:
Yep I just fixed this.